typegate/src/typegate/register.ts

```diff
@@ -18,18 +19,8 @@ export interface Migrations {
   migrations: string;
 }
 
-export interface RegistrationResult {
-  typegraphName: string;
-  messages: Array<MessageEntry>;
-  migrations: Array<Migrations>;
-  resetRequired: Array<string>;
-}
-
 export abstract class Register {
-  abstract set(
-    payload: string,
-    secrets: Record<string, string>,
-  ): Promise<RegistrationResult>;
+  abstract add(engine: Engine): Promise<void>;
```

Comment:
the register interface with these signatures looks a bit suspicious or artificial? is there a better interface possible?

Reply:
Do you mean like: `register(engine.name, engine)`?

Reply:
yes, at the end of the day, maybe even remove the abstraction as it is a simple map?